### PR TITLE
[feature] demo of a smart search

### PR DESCRIPTION
**Pull request by ponsfrilus**
This commit bring a new search bar that allow users to enter keyword to  choose type of search. For now it is hard coded and inclues:
* `?` for Google
* `g` for Go EPFL
* `m` for EPFL Map
* `p` for EPFL People
* `sn` for ServiceNow
* `u` for EPFL Units

The <kbd>Esc</kbd> key clear the search input.

This still need some work but can be used as POC.

smart-search

### CODE REVIEW

**Comment by williambelle** [comment]
It works well.

We should add a dropdown for the choice of the search engine:

Screenshot from 2023-11-02 18-35-16

File: src/public/js/search.js [hunk 1]
```diff
@@ -1,3 +1,103 @@
-$('#search-input').on('input', function () {
-  console.log($(this).val());
+const setSearchOn = (searchEngine, keepValue = '') => {
```
Comment (williambelle) [inline src/public/js/search.js:1]: `$(function(){ ... });`  to be ready ?

File: src/public/js/search.js [hunk 2]
```diff
@@ -1,3 +1,103 @@
...
+    window.open('https://search.epfl.ch/?q=' + terms);
+  }
+  if (searchEngine === 'EPFL Map') {
+    window.open('https://plan.epfl.ch/?room=' + terms);
+  }
+  if (searchEngine === 'EPFL People') {
+    window.open('https://search.epfl.ch/?filter=people&q=' + terms);
+  }
+  if (searchEngine === 'EPFL Units') {
+    window.open('https://search.epfl.ch/?filter=units&q=' + terms);
```
Comment (williambelle) [inline src/public/js/search.js]: `filter=unit` (without s)

File: src/views/pages/home.ejs [hunk 3]
```diff
@@ -1,32 +1,90 @@
...
-                class="form-control form-control-search"
-              />
-              <svg aria-hidden="true" class="icon">
-                <use xlink:href="#icon-search"></use>
-              </svg>
-            </p>
+      <div class="welcome">EPFL Start page</div>
+      <hr />
+      <div class="col-auto">
+        <label class="sr-only" for="inlineFormInputGroup">Username</label>
```
Comment (williambelle) [inline src/views/pages/home.ejs]: `for="dynamic_search"` (based on the input below).

Username ?

File: src/public/js/search.js [hunk 4]
```diff
@@ -1,3 +1,103 @@
...
+  }
+  if (searchEngine === 'ServiceNow') {
+    window.open(
+      // ↓↓↓ yes this sucks
+      // eslint-disable-next-line
```
Comment (williambelle) [inline src/public/js/search.js]: It works but probably not the best approach.

File: src/public/js/search.js [hunk 5]
```diff
@@ -1,3 +1,103 @@
-$('#search-input').on('input', function () {
-  console.log($(this).val());
+const setSearchOn = (searchEngine, keepValue = '') => {
+  $('#search_on').html(searchEngine);
```
Comment (williambelle) [inline src/public/js/search.js]: `.text()`

File: src/views/pages/home.ejs [hunk 6]
```diff
@@ -1,32 +1,90 @@
...
+                Wikipedia</a
+              >
+            </div>
           </div>
+          <input
+            id="dynamic_search"
+            type="text"
+            class="form-control"
+            placeholder="Search…"
+          />
```
Comment (williambelle) [inline src/views/pages/home.ejs:84]: Do we want the magnifying glass icon in the input?

**Review by williambelle — COMMENTED** [review]
Nice to have for a future version:

- As a user, I want to choose the default search engine.
- As a user, I want to choose the letter(s) associated with a search engine.
- As a user, I want to choose the key that triggers the choice of a search engine.

File: src/public/js/search.js (hunk 1 above)
Comment (ponsfrilus) [reply src/public/js/search.js:1]: Done.

File: src/public/js/search.js (hunk 2 above)
Comment (ponsfrilus) [reply src/public/js/search.js]: Done.

File: src/views/pages/home.ejs (hunk 3 above)
Comment (ponsfrilus) [reply src/views/pages/home.ejs]: Done.

File: src/public/js/search.js (hunk 4 above)
Comment (ponsfrilus) [reply src/public/js/search.js]: ¯\\_(ツ)_/¯ 80 CPL limit doesn't work well with URL

File: src/public/js/search.js (hunk 5 above)
Comment (ponsfrilus) [reply src/public/js/search.js]: Done.

File: src/views/pages/home.ejs (hunk 6 above)
Comment (ponsfrilus) [reply src/views/pages/home.ejs:84]: Maybe in another issue ?

File: src/public/js/search.js (hunk 4 above)
Comment (ponsfrilus) [reply src/public/js/search.js]: `yes this sucks` comment removed.

File: src/views/pages/home.ejs (hunk 6 above)
Comment (williambelle) [reply src/views/pages/home.ejs:84]: Yep

**Review by williambelle — APPROVED** [review]

**Pull request merged**